dask/array/reductions.py

```diff
@@ -145,16 +145,23 @@ def reduction(x, chunk, aggregate, axis=None, keepdims=False, dtype=None,
     tmp = blockwise(chunk, inds, x, inds, axis=axis, keepdims=True, dtype=x.dtype)
     tmp._chunks = tuple((output_size, ) * len(c) if i in axis else c
                         for i, c in enumerate(tmp.chunks))
+
+    reduced_meta = None
+    if hasattr(x, '_meta'):
+        reduced_meta = blockwise(chunk, inds, x._meta, inds, axis=axis,
+                                 keepdims=True, dtype=x.dtype)
```

Comment:
I'm confused about the objective of `reduced_meta` here.  Can you help me understand this?

Also, do we need to call `blockwise`?  Or are you just calling blockwise in order to get the `meta` object?

Reply:
I'm trying to ensure that `_meta` goes through all the same steps as the main array, since I assume it's not possible to predict how every single step may change the arrays and their data type.

Reply:
Can we use `tmp._meta` instead here perhaps?

Reply:
In general calling compute can be quite complex to do. We engage the full optimization system.  If we're using the distributed scheudler then we trigger network communications.  If we're using Dask in an asynchronous mode then things will probably block in bad ways.  If we need to call compute then we can turn all of this stuff off during that compute call, but I'd prefer to avoid it if we can.

Reply:
I totally get your point now @mrocklin. However, I don't immediately see a different way here. What I could do is not "reduce" `_meta`, but I presume that could lead to different results for complex types. Do you have any suggestions on how to better handle it here?